components/region_cache_memory_engine/src/background.rs

```diff
@@ -154,10 +167,16 @@ impl BackgroundRunner {
         Self { engine_core }
     }
 
-    fn ranges_for_gc(&self) -> Vec<CacheRange> {
-        let mut core = self.engine_core.write().unwrap();
-        let ranges: Vec<CacheRange> = core.range_manager().ranges().keys().cloned().collect();
-        core.set_ranges_gcing(ranges.clone());
+    fn ranges_for_gc(&self) -> BTreeSet<CacheRange> {
+        let ranges: BTreeSet<CacheRange> = {
+            let core = self.engine_core.read().unwrap();
+            core.range_manager().ranges().keys().cloned().collect()
+        };
+        let ranges_clone = ranges.clone();
+        {
+            let mut core = self.engine_core.write().unwrap();
+            core.mut_range_manager().set_ranges_in_gc(ranges_clone);
```

Comment:
It seems to me ranges_in_gc is always same as ranges. Is there any reason to have it separately?